torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -1310,12 +1310,58 @@ void traverseProfileIValues(
   }
 }
 
+// break `linear` layer into `matmul` and `add_optional`. This allows us to fuse
+// the binary operation without supporting gemm.
+// Note that we are not breaking `linear` layer without bias.
+void decomposeLinearOps(Block* block) {
+  std::vector<Node*> linear_nodes;
+  for (Node* n : block->nodes()) {
+    for (Block* b : n->blocks()) {
+      decomposeLinearOps(b);
+    }
+    // only decompose `linear` layer with bias.
+    if (n->kind() == aten::linear &&
+        !n->input(2)->type()->isSubtypeOf(
+            static_cast<c10::TypePtr>(NoneType::get()))) {
+      linear_nodes.push_back(n);
+    }
+  }
+
+  auto graph = block->owningGraph();
+  for (Node* n : linear_nodes) {
+    WithInsertPoint guard(n);
+    auto weight_t = graph->insertNode(graph->create(aten::t, {n->input(1)}, 1));
+    auto matmul = graph->insertNode(
+        graph->create(aten::matmul, {n->input(0), weight_t->output()}, 1));
+    auto input_tensor_type = n->input(0)->type()->cast<c10::TensorType>();
+    auto mat0_size = input_tensor_type->sizes().concrete_sizes();
+    auto mat1_size =
+        n->input(1)->type()->cast<c10::TensorType>()->sizes().concrete_sizes();
```

Comment:
Are these actually concrete sizes? Or this is this related to inserting profiling nodes?

Reply:
This is propagating profiled sizes. So they are concrete but it shouldn't break our dynamic size support